### ISSUE TITLE
codegen - QCodeGen::GenerateAggregate never called

### ISSUE THREAD

**Issue by brynjaro** [qcubed/qcubed#1249]
I just installed version 3.0.8 via Composer (Ubuntu 16.04, apache 2.4.18, MySQL 5.7.17). I am code generating against database with a lot of tables, and some of them give me Warnings like:

`The following errors were reported:
Warning: TypeTable xxx_type contains a type name which is a reserved word: new.  Appended _ to the beginning of it.`

Because of this errors (or actually warnings) the code in line 75 in codegen.php never gets called and just ignore the call to `QCodeGen::GenerateAggregate()` without any warning.

`if (!$strErrors) {
	foreach (QCodeGen::GenerateAggregate() as $strMessage) { ?>
		<p><strong><?php _p($strMessage); ?></strong></p>
	<?php }
} ?>`

In my case, I just have to comment out the `if (!$strErrors) {` condition and everything works fine.

I have been running QCodo version 2.0.2 for ages and this  condition is not there and I have not run into any problems. Maybe this is too strict? Do we have case when we can't run this code at all? And if so, some warning would be nice.

Please let me know if you need more information or if I can help.

**Comment by spekary** [comment]
This means one of the names you are using in the type table is "new". If QCubed tries to generate this, then it will try to create a constant called "new" and your php will not run. 

What did v2 generate in this case? What do you propose should happen in this case?

**Comment by brynjaro** [comment]
v2 generated `const _New = 1;` (appending _ to the beginning) and v3 is doing exactly the same if you remove the `if (!$strErrors) {` condition and let the codegen do it's work. This is in fact no error, just warning and QCubed can handle it really well.

**Comment by spekary** [comment]
Sounds good. But, we need to have a look at why we are preventing the GenerateAggregate from functioning. I suspect there were situations where this was causing codegen to fail in bad ways, like without generating anything. You will need to do a Git Blame and see what the comments suggested as to why we were doing this. Perhaps we can separate $strWarnings from $strErrors.

**Comment by matthiaz** [comment]
@spekary I did a git history on it. You changed it in this commit: 
https://github.com/qcubed/qcubed/commit/777296dc38d82c188a4568a5f18e3af7ff7ed653

Producing intelligible error message if the user tries to codegen before doing anything to the configuration file.
I suggest removing the if(!$strErrors) but leaving the database if statements you added. Since @brynjaro  didn't submit a pull request, I did. 
Feel free to decline if I did miss something though... 

https://github.com/qcubed/qcubed/pull/1296
